### PR TITLE
autoscaler attach/detatch

### PR DESCRIPTION
**Pull request by shaiguitar**
Ran this against a live amazon account and this did the expected. 

Amazon doc links inline in the commit.

### CODE REVIEW

File: lib/fog/aws/requests/auto_scaling/attach_instances.rb [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        # `aws autoscaling attach-instances --instance-ids i-93633f9b --auto-scaling-group-name my-auto-scaling-group`
+        #
+        # ==== Parameters
+        #
+        #   * AutoScalingGroupName<~String> - The name of the Auto Scaling group``
+        #   * 'InstanceIds'<~Array> - The list of Auto Scaling instances to detach.
+        #
+        # ==== See Also
+        #
+        # http://docs.aws.amazon.com/cli/latest/reference/autoscaling/attach-instances.html
```
Comment (lanej) [inline lib/fog/aws/requests/auto_scaling/attach_instances.rb]: should be http://docs.aws.amazon.com/AutoScaling/latest/APIReference/API_AttachInstances.html

File: lib/fog/aws/requests/auto_scaling/detach_instances.rb [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        # ==== Parameters
+        #
+        #   * AutoScalingGroupName<~String> - The name of the Auto Scaling group``
+        #   * 'InstanceIds'<~Array> - The list of Auto Scaling instances to detach.
+        #
+        # * options<~Hash>:
+        # 'shouldDecrementDesiredCapacity'<~Boolean> - decrement the asg capacity or not (it will boot another if an instance id detached)
+        #
+        # ==== See Also
+        # http://docs.aws.amazon.com/AutoScaling/latest/DeveloperGuide/detach-instance-asg.html
```
Comment (lanej) [inline lib/fog/aws/requests/auto_scaling/detach_instances.rb]: http://docs.aws.amazon.com/AutoScaling/latest/APIReference/API_DetachInstances.html

**Comment by shaiguitar** [comment]
pushed doc link change

File: lib/fog/aws/version.rb [hunk 3]
```diff
@@ -1,5 +1,5 @@
 module Fog
   module AWS
-    VERSION = "0.8.1"
+    VERSION = "0.9.0"
```
Comment (lanej) [inline lib/fog/aws/version.rb]: i'll worry about the version bump, tagging, etc.

**Comment by shaiguitar** [comment]
reverted lib/version.rb change to what it was

**Comment by geemus** [comment]
@lanej looks like you are on top of this, but just let me know if you need anything.

**Comment by lanej** [comment]
@geemus thanks

**Comment by lanej** [comment]
@shaiguitar the mocks don't manipulate any data.  is that intentional ?

**Comment by shaiguitar** [comment]
@lanej Basically just copied over what was in https://github.com/fog/fog-aws/blob/master/lib/fog/aws/requests/auto_scaling/attach_load_balancers.rb#L36-L55 . There's no documentation/wiki on adding mock behavior and it's been a while so I'm a little hazy on the idiomatics.

Seems like the asg mock interactions are broken as well with regards to the other instance manipulation requests, see these two (`Fog::Mock.not_implemented`) `lib/fog/aws/requests/auto_scaling/set_instance_health.rb` and `lib/fog/aws/requests/auto_scaling/terminate_instance_in_auto_scaling_group.rb`

In the case of `lib/fog/aws/requests/auto_scaling/set_desired_capacity.rb` if we reload after setting desired capacity it should also create some instances, but nothing is happening in the mock either.

There's probably some other examples but that should be an indication it's not quite there for the asgs anyway.

So, I can add another `not_implemented` in there or I can take a look at implementing these mocks, but I'll need some hand holding and or docs/wiki.

**Comment by shaiguitar** [comment]
Bump. The real implementation should work. I can add in `not_implemented` if that's better than the current alternative

**Comment by shaiguitar** [comment]
Anything I can do to get this released?

**Comment by lanej** [comment]
@shaiguitar rebase for a passing build

**Comment by shaiguitar** [comment]
@lanej green

**Comment by shaiguitar** [comment]
Thanks!

**Comment by lanej** [comment]
this commit screwed with the versions @2456533f029698f060f4c368af22d9106d74b64f.  Next time remove the commit instead of reverting in the same branch.

**Pull request merged**